src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/OwnerNames.kt

```diff
@@ -10,21 +10,53 @@
 
 package org.jetbrains.kotlinx.lincheck.strategy.managed
 
+import org.jetbrains.lincheck.descriptors.*
+import org.jetbrains.lincheck.analysis.*
+import org.jetbrains.kotlinx.lincheck.util.*
+import org.jetbrains.lincheck.trace.isThisName
 import org.jetbrains.kotlinx.lincheck.transformation.toSimpleClassName
-import org.jetbrains.kotlinx.lincheck.util.AtomicApiKind
-import org.jetbrains.kotlinx.lincheck.util.AtomicMethodDescriptor
-import org.jetbrains.kotlinx.lincheck.util.getAtomicArrayAccessInfo
-import org.jetbrains.kotlinx.lincheck.util.getAtomicFieldUpdaterAccessInfo
-import org.jetbrains.kotlinx.lincheck.util.getUnsafeAccessInfo
-import org.jetbrains.kotlinx.lincheck.util.getVarHandleAccessInfo
-import org.jetbrains.lincheck.analysis.ShadowStackFrame
-import org.jetbrains.lincheck.analysis.findCurrentReceiverFieldReferringTo
-import org.jetbrains.lincheck.analysis.findLocalVariableFieldReferringTo
-import org.jetbrains.lincheck.analysis.findLocalVariableReferringTo
-import org.jetbrains.lincheck.analysis.isThisName
-import org.jetbrains.lincheck.descriptors.AccessLocation
-import org.jetbrains.lincheck.descriptors.ArrayElementByIndexAccessLocation
-import org.jetbrains.lincheck.descriptors.FieldAccessLocation
+
+internal fun findOwnerName(
+    className: String,
+    shadowStackFrame: ShadowStackFrame,
+): String? {
+    // If the class name matches the current instance's class in the shadow stack,
+    // return null since the field belongs to the current class context and doesn't
+    // need an explicit owner prefix
+    if (className == shadowStackFrame.instance?.javaClass?.name) {
+        return null
+    }
+    return className.toSimpleClassName()
+}
+
+fun findOwnerName(
+    obj: Any?,
+    className: String,
+    codeLocationId: Int,
+    shadowStackFrame: ShadowStackFrame,
+    objectTracker: ObjectTracker,
+): String? {
+    if (obj == null) {
+        return findOwnerName(className, shadowStackFrame)
+    }
+    // do not prettify thread names
```

Comment:
Why?

Reply:
> Should we show both the variable name and object names for threads

Maybe, but then what is so special about thread objects in particular? I guess in this case we would like to apply this technique to all objects?